MCP Server API (user) documentation (#2302)
ansible/vscode-ansible: docs/mcp/README.md

## Features

The Ansible MCP Server provides the following capabilities:

### 📚 Information & Documentation

- **Zen of Ansible**: Access Ansible's design philosophy and principles
- **Best Practices**: Get comprehensive guidelines for writing quality Ansible content
- **Tool Discovery**: List all available MCP tools and their capabilities

### 🔧 Environment Management

- **Environment Info**: Check Python, Ansible, and development tool versions
- **Setup Automation**: Automatically configure virtual environments and install dependencies
- **Tool Installation**: Install and verify Ansible Development Tools (ADT)

### 🏗️ Project Scaffolding

- **Playbook Creation**: Generate new Ansible playbooks with proper structure
- **Collection Creation**: Scaffold new Ansible collections with best practices

### ✅ Code Quality

- **Ansible Lint**: Automated linting with fix capabilities
- **Execution Environment Builder**: Create and validate execution environment definitions

### ▶️ Playbook Execution

- **Ansible Navigator**: Execute playbooks with smart environment detection and container management

## Getting Started

> The Ansible MCP Server is currently available for testing and development. It runs as a child process and communicates via stdio transport.

Reviewer: ```suggestion
The Ansible MCP Server is currently available as a technical preview.

The server runs as a child process and communicates via stdio transport.
```

We should also have info on starting the server here.
